src/python.rs

```diff
@@ -162,30 +162,33 @@ struct StringCache;
 
 impl StringMaybeCache for StringCache {
     fn get<'py>(py: Python<'py>, json_str: &str) -> Py2<'py, PyAny> {
-        static STRINGS_CACHE: GILOnceCell<GILProtected<RefCell<AHashMap<String, PyObject>>>> = GILOnceCell::new();
+        static STRINGS_CACHE: GILOnceCell<GILProtected<RefCell<HashMap<String, PyObject>>>> = GILOnceCell::new();
 
         // from tests, 0 and 1 character strings are faster not cached
         if (2..64).contains(&json_str.len()) {
             let cache = STRINGS_CACHE
-                .get_or_init(py, || GILProtected::new(RefCell::new(AHashMap::new())))
+                .get_or_init(py, || GILProtected::new(RefCell::new(HashMap::new())))
                 .get(py);
 
-            // Finish the borrow before matching, so that the RefCell isn't borrowed for the whole match.
-            let key = cache.borrow().get(json_str).map(|key| key.clone_ref(py));
-
-            match key {
-                Some(key) => key.attach_into(py),
-                None => {
-                    let py_key = PyString::new2(py, json_str);
-                    let mut cache_writable = cache.borrow_mut();
-                    // 500k limit means 1m keys + values, 1m 64 byte strings is ~64mb
-                    if cache_writable.len() > 500_000 {
-                        cache_writable.clear();
-                    }
-                    cache_writable.insert(json_str.to_owned(), py_key.to_object(py));
-                    py_key.into_any()
+            let mut map = cache.borrow_mut();
+            let entry = map.raw_entry_mut().from_key(json_str);
+
+            let (py_string, inserted) = match entry {
+                RawEntryMut::Vacant(view) => {
+                    let py_string = PyString::new2(py, json_str);
+                    view.insert(json_str.to_owned(), py_string.to_object(py));
+                    (py_string.into_any(), true)
+                }
+                RawEntryMut::Occupied(view) => (view.get().clone_ref(py).attach_into(py), false),
+            };
+            if inserted {
+                // 500k limit means 1m keys + values, 1m 64 byte strings is ~64mb
```

Comment:
better I think do only do this if we inserted a key rather than always, even if that means clearing the key we just inserted.